### PR TITLE
[Kotlin] Ability to generate sealed interfaces

### PR DESCRIPTION
**Pull request by kbrooks**
---

### Description

Adds `sealed` keyword to generated interfaces and unions in order to enable exhaustive type-checking in written code. Only works in and in Kotlin, even though Scala has support for sealed interfaces, because Scala has the restriction that classes inheriting a sealed interface are in the same file as that interface. Java has support for sealed interfaces but requires a `permits A, B, ...` syntax that makes it more difficult to support since we would need to parse the entire schema for all values that implement a given interface.

Related to #868 

---

Changes were made to:
- [x] Codegen library - Java
- [x] Codegen library - Kotlin
- [ ] Codegen library - Scala
- [x] Maven plugin
- [x] Gradle plugin
- [ ] SBT plugin

### CODE REVIEW

**Comment by kbrooks** [comment]
Sonar is failing for mysterious reasons

File: docs/codegen-options.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ See [DirectiveAnnotationsMapping](#option-directiveannotationsmapping)* |
...
 | `generateModelOpenClasses`                            | Boolean                                                               | false                                                  | The class type of the generated model. If true, generate normal classes, else generate data classes. It only support in kotlin(```data class```) and scala(```case class```). Maybe we will consider to support Java ```record``` in the future.|
 | `initializeNullableTypes`                             | Boolean                                                               | false                                                  | Adds a default null value to nullable arguments. Only supported in Kotlin.
+| `generateSealedInterfaces`                            | Boolean                                                               | false                                                  | This applies to generated interfaces on unions and interfaces in Kotlin. If true, generate sealed interfaces, else generate normal ones. It is only supported in kotlin. |
```
Comment (kbrooks) [inline docs/codegen-options.md:60]:
```suggestion
| `generateSealedInterfaces`                            | Boolean                                                               | false                                                  | This applies to generated interfaces on unions and interfaces. If true, generate sealed interfaces, else generate normal ones. It is only supported in Kotlin. |
```

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenKeys.scala [hunk 2]
```diff
@@ -124,6 +124,8 @@ trait GraphQLCodegenKeys {
 
   val generateModelOpenClasses = settingKey[Boolean]("The class type of the generated model. If true, generate normal classes, else generate case class.")
 
+  val generateSealedInterfaces = settingKey[Boolean]("If true, generate sealed interfaces for GraphQL unions and interface, else generate normal interfaces.")
```
Comment (kbrooks) [inline plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/GraphQLCodegenKeys.scala:127]:
```suggestion
  val generateSealedInterfaces = settingKey[Boolean]("If true, generate sealed interfaces for GraphQL unions and interfaces, else generate normal interfaces.")
```

**Comment by kbrooks** [comment]
java interface is missing "permits"

**Review by kobylynskyi — APPROVED** [review]
Looks good!

**Pull request merged**